### PR TITLE
Anonymous Posting Implementation

### LINKED ISSUE #16: Connect Frontend & Backend of isAnonymous
**User Story:** As a student, I want to make a post anonymously, so that I can ask a question without revealing my identity.

**Estimated effort:** 8 Hours

**Dependencies:** None

**Initial assignments:** @wenchaohu1 @aqimeral @22zhuamanda 

**Milestones:** Complete by the first week of Sprint 2 if dependencies are achieved.

**Acceptance criteria:**
Using the checkbox when editing or creating a post will cause the user ID for the post to be replaced by "Anonymous"

### PR DESCRIPTION
**Pull request by 22zhuamanda**
Solves issue #16

**Content added:**
Sync plugin;
Initialized static isAnonymous in database;
Initialized anonymizeCheckbox const, dynamically call anonymizeCheckbox.checked;
Linked composer.tpl to composer.js;
Changed username display according to isAnonymous value.

**Current Progress:**
1. Users are able to toggle the anonymize checkbox
2. When it is checked, display 'Anonymous' in place of the user's actual username

**Screenshots:**
Screenshot 2024-02-29 at 12 19 46 AM
For this first post, do not check the anonymize button.
Screenshot 2024-02-29 at 12 20 08 AM
For this second post, check the anonymize button.
Screenshot 2024-02-29 at 12 21 07 AM copy
When running redis-cli, you can see that for the first post, isAnonymous is false, and for the second post, isAnonymous is true.
Screenshot 2024-02-29 at 12 34 44 AM
After the second post is posted, the username displays as 'Anonymous' instead of the user's actual username.

**Strengths**
Preserve anonymity of users when desired.

**Weaknesses**
Cannot differentiate between multiple anonymous users (all are named 'Anonymous').

**Next Steps**
Write test cases for this feature, update documentation to reflect updated user guide.

### CODE REVIEW

File: UserGuide.md
Comment (FireACNC) [inline UserGuide.md:1]: Nice instructions!

File: plugins/nodebb-plugin-composer/library.js
Comment (FireACNC) [inline plugins/nodebb-plugin-composer/library.js:1]: Would be better to comment out/delete debugging print statements!

File: plugins/nodebb-plugin-composer/static/templates/composer.tpl
Comment (FireACNC) [inline plugins/nodebb-plugin-composer/static/templates/composer.tpl:1]: Great style on adding the id and name!

File: test/controllers.js
Comment (FireACNC) [inline test/controllers.js:1]: would be better to leave a comment saying why we change the status code!

**Comment by 22zhuamanda** [comment]
UPDATE: added user guide documentation

**Review by FireACNC — APPROVED** [review]
LGTM!

File: plugins/nodebb-plugin-composer/library.js
Comment (wenchaohu1) [reply plugins/nodebb-plugin-composer/library.js:1]: Thank you very much, just removed the print statements!

**Pull request merged**